Dockerfile

```diff
@@ -1,69 +1,46 @@
+
+# In contrast to the runtime image below, here we can afford to install additional
+# tools and recommended packages. But this image gets pushed to a registry by the CI as a cache,
+# so it still makes sense to keep this layer small by removing /var/cache.
+RUN ${APT_GET} update && \
+    ${APT_GET} install -y gcc libndctl-dev make git curl iproute2 pkg-config xfsprogs e2fsprogs parted openssh-client python3 python3-venv && \
+    rm -rf /var/cache/*
 RUN curl -L https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz | tar -zxf - -C / && \
     mkdir -p /usr/local/bin/ && \
     for i in /go/bin/*; do ln -s $i /usr/local/bin/; done
 
-WORKDIR /
-RUN curl --fail --location --remote-name https://github.com/pmem/ndctl/archive/v${NDCTL_VERSION}.tar.gz
-RUN tar zxvf v${NDCTL_VERSION}.tar.gz && mv ndctl-${NDCTL_VERSION} ndctl
-WORKDIR /ndctl
-RUN ./autogen.sh
-# We install into /usr/local (keeps content separate from OS) but
-# then symlink the .pc files to ensure that they are found without
-# having to set PKG_CONFIG_PATH. The .pc file doesn't contain an -rpath
-# and thus linked binaries do not find the shared libs unless we
-# also symlink those.
-RUN ./configure --prefix=/usr/local ${NDCTL_CONFIGFLAGS}
-RUN make install && \
-    ln -s /usr/local/lib/pkgconfig/libndctl.pc /usr/lib64/pkgconfig/ && \
-    ln -s /usr/local/lib/pkgconfig/libdaxctl.pc /usr/lib64/pkgconfig/ && \
-    for i in /usr/local/lib/lib*.so.*; do ln -s $i /usr/lib64; done
-
-# The source archive has no license file. We link to the copy in GitHub instead.
-RUN echo "For source code and licensing of ndctl, see https://github.com/pmem/ndctl/blob/v${NDCTL_VERSION}/COPYING" >/usr/local/lib/NDCTL.COPYING
-
 # Clean image for deploying PMEM-CSI.
-FROM ${CLEAR_LINUX_BASE} as runtime
-ARG CLEAR_LINUX_BASE
-ARG SWUPD_UPDATE_ARG
+FROM ${LINUX_BASE} as runtime
+ARG APT_GET="env DEBIAN_FRONTEND=noninteractive apt-get"
+ARG CACHEBUST
 ARG BIN_SUFFIX
 LABEL maintainers="Intel"
 LABEL description="PMEM CSI Driver"
 
-# update and install needed bundles:
+# Update and install the minimal amount of additional packages that
+# are needed at runtime:
 # file - driver uses file utility to determine filesystem type
-# xfsprogs - XFS filesystem utilities
-# storge-utils - for lvm2 and ext4(e2fsprogs) utilities
-ARG CACHEBUST
-RUN echo "Updating runtime image from ${CLEAR_LINUX_BASE} to ${SWUPD_UPDATE_ARG:-the latest release}."
-RUN swupd update ${SWUPD_UPDATE_ARG} && swupd bundle-add file xfsprogs storage-utils \
-    $(if [ "$BIN_SUFFIX" = "-test" ]; then echo fio; fi) && \
-    rm -rf /var/lib/swupd /var/tmp/swupd
+# xfsprogs, e2fsprogs - formating filesystems
+# lvm2 - volume management
+# ndctl - pulls in the necessary library, useful by itself
+# fio - only included in testing images
+RUN ${APT_GET} update && \
+    ${APT_GET} upgrade -y --no-install-recommends && \
+    ${APT_GET} install -y --no-install-recommends file xfsprogs e2fsprogs lvm2 ndctl \
```

Comment:
So, now on we no longer building `ndctl` from the source.

Reply:
Correct. It makes our bill of materials and Dockerfile simpler. I don't think we ever really needed the ability to build a newer version than the one offered by the distro, and as ndctl is even more mature now I don't expect that either.